SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -120,6 +136,58 @@ void ASpatialDebugger::BeginPlay()
 	}
 }
 
+void ASpatialDebugger::OnAuthorityGained()
+{
+	if (NetDriver->LoadBalanceStrategy)
+	{
+		if (UGridBasedLBStrategy* GridBasedLBStrategy = Cast<UGridBasedLBStrategy>(NetDriver->LoadBalanceStrategy))
+		{
+			const UGridBasedLBStrategy::LBStrategyRegions LBStrategyRegions = GridBasedLBStrategy->GetLBStrategyRegions();
+			WorkerRegions.Empty();
+			for (int i = 0; i < LBStrategyRegions.Num(); i++)
```

Comment:
Nit: I think unreal would prefer using a for each style loop. Would kind of make things subtly easier to read.

Reply:
Changing the `.Empty()` call above to instead pre-allocate the correct value means that instead of calling `WorkerRegions.Add(...)` below I just want to index the element appropriately, i.e. `WorkerRegions[i] = ...` meaning I think sticking with the index is necessary rather than for each - which do you think is preferable?

Reply:
Oooh very sexy. Yeah I like that. The only tiny tiny thing that some people might argue for is then caching a ref to the element at the start of the loop instead of accessing it twice.